### PR TITLE
Add SMZ3 support

### PR DESCRIPTION
**Pull request by lordlou**
Made a python port of Total SMZ3 Randomizer: https://github.com/tewtal/SMZ3Randomizer
This is allowed if we keep features and logic the same as the original, excluding AP's generic features.
Also uses a modified version of SMZ3 basepatch: https://github.com/lordlou/alttp_sm_combo_randomizer_rom

### CODE REVIEW

File: host.yaml [hunk 1]
```diff
@@ -111,3 +111,14 @@ oot_options:
 soe_options:
   # File name of the SoE US ROM
   rom_file: "Secret of Evermore (USA).sfc"
+smz3_options:
```
Comment (Berserker66) [inline host.yaml:114]: Any chance to pull these from the SM and LttP sections?

**Review by Ijwu — CHANGES_REQUESTED** [review]
We'll need setup guides and an addition to the game info page before I think this is live ready.

File: worlds/smz3/Options.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class SwordLocation(Choice):
+    displayname = "Sword Location"
+    option_Randomized = 0
+    option_Early = 1
+    option_Uncle = 2
+    default = 0
+
+class MorphLocation(Choice):
+    displayname = "Morph Location"
```
Comment (Berserker66) [inline worlds/smz3/Options.py:18]: the attribute for display name was changed to `display_name`, to comply with pep8

File: WebHostLib/static/assets/tutorial/smz3/multiworld_en.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    1. Extract your emulator's folder to your Desktop, or somewhere you will remember.
+    2. Right-click on a ROM file and select **Open with...**
+    3. Check the box next to **Always use this app to open .sfc files**
+    4. Scroll to the bottom of the list and click the grey text **Look for another App on this PC**
+    5. Browse for your emulator's `.exe` file and click **Open**. This file should be located inside the folder you
+       extracted in step one.
+
+### Macintosh Setup
+
+- We need volunteers to help fill this section! Please contact **Farrak Kilhn** on Discord if you want to help.
```
Comment (Ijwu) [inline WebHostLib/static/assets/tutorial/smz3/multiworld_en.md:37]: I'm not sure it's right to leave this copy-pasta in if Farrak didn't actually write this guide. Maybe remove the section or replace with your own name if you're interested in being point of contact for mac setup.

Comment (lordlou) [reply WebHostLib/static/assets/tutorial/smz3/multiworld_en.md]: Oops, that one slipped. I removed the Mac section.

**Review by Ijwu — APPROVED** [review]
Docs seem good.

**Pull request merged**